nym-vpn-core/nym-vpn-lib/src/mobile/ios/default_path_observer.rs

```diff
@@ -0,0 +1,64 @@
+use tokio::sync::mpsc::{UnboundedReceiver, UnboundedSender};
+
+/// The type alias for the receiving end of path observer.
+pub type DefaultPathReceiver = UnboundedReceiver<OSDefaultPath>;
+
+/// Observer type that wraps network path changes into a channel.
+#[derive(Debug)]
+pub struct DefaultPathObserver {
+    tx: UnboundedSender<OSDefaultPath>,
+}
+
+impl DefaultPathObserver {
+    pub fn new(tx: UnboundedSender<OSDefaultPath>) -> Self {
+        Self { tx }
+    }
+}
+
+impl OSDefaultPathObserver for DefaultPathObserver {
+    fn on_default_path_change(&self, new_path: OSDefaultPath) {
+        if self.tx.send(new_path).is_err() {
+            tracing::warn!("Failed to send default path change.");
+        }
+    }
+}
+
+#[derive(uniffi::Enum, Debug)]
```

Comment:
What's the thinking when deciding between placing stuff in this mod vs `platform`? Is `platform` something that will be phased out in time?

Reply:
Simply a logical grouping of related types. Everything under `mobile/ios` in essence is a platform-specific stuff. I didn't want to move things too apart since it breaks connection between types.